doc/dev_guide/internals/box_protocol.rst

```diff
@@ -800,6 +803,41 @@ The body is a 2-item map:
         IPROTO_LSN: :samp:`{{MP_INT integer}}`
     })
 
+..  _box_protocol-id:
+
+IPROTO_ID = 0x49
+~~~~~~~~~~~~~~~~~~~~~~
+
+Clients send this message to inform the server about the protocol version and
+features they support. Based on this information, the server can limit its
+functionality when interacting with these clients.
+
+The body is a 2-item map:
+
+..  cssclass:: highlight
+..  parsed-literal::
+
+    # <size>
+    msgpack(:samp:`{{MP_UINT unsigned integer = size(<header>) + size(<body>)}}`)
+    # <header>
+    msgpack({
+        IPROTO_REQUEST_TYPE: IPROTO_ID,
+        IPROTO_SYNC: :samp:`{{MP_UINT unsigned integer}}`
+    })
+    # <body>
+    msgpack({
+        IPROTO_VERSION: :samp:`{{MP_UINT unsigned integer}}}`,
+        IPROTO_FEATURES: :samp:`{{MP_ARRAY array of unsigned integers}}}`
+    })
+
+IPROTO_VERSION is the integer number that increases by one with each change in
+the IPROTO_FEATURES set. The latest version is |iproto_version|.
+
+Supported IPROTO_FEATURES are listed in the
```

Comment:
@locker do you think we should provide the features list explicitly? It can make the doc clearer but will require additional support in the future

Reply:
I think they should be documented here. Referring to the code isn't a good idea, because the comments there are intended for internal developers. Besides, we can rename the file in future.

Reply:
Added the descriptions. Ready for review